cellmlmanip/model.py

```diff
@@ -410,27 +410,46 @@ def get_equation_graph(self, refresh=False) -> nx.DiGraph:
         self.graph = graph
         return graph
 
-    def get_equations_for(self, symbols, lexicographical_sort=True):
+    def get_equations_for(self, symbols, sort_by_input_symbols=False, excluded_symbols=[]):
         """Get all equations for given collection of symbols
 
         Results are sorted in topographical order.
         :param symbols: the symbols to get the equations for
-        :param lexicographical_sort: indicates whether the result is sorted in lexicographical order first
+        :param sort_by_input_symbols: indicates whether the sorting in symbols determines the sorting of the output
```

Comment:
Doing this manually just feels wrong. Surely there's something in the NetworkX library we can get to do this for us?

For instance, the [lexicographic](https://networkx.github.io/documentation/stable/reference/algorithms/generated/networkx.algorithms.dag.lexicographical_topological_sort.html) sort takes a `key` argument to resolve ambiguities in the sort order. Would that work?

Reply:
What you'd have to do is write a key function key such that 
given input [x1...xN] to the get_equations_for fuction
key(x1) < ... < ... key(xN) (which is straightforward) but also
key (y1) < ... < key(yN) where yn is a parent of xn

I don't really see any way of doing this without actually doing the sort unless I'm missing something?